PyISY/Events/events.py

```diff
@@ -62,22 +61,27 @@ def _routemsg(self, msg):
         self.parent.log.debug('ISY Update Received:\n' + msg)
 
         # direct the event message
-        cntrl = '<control>{0}</control>'
-        if cntrl.format('_0') in msg:  # ISY HEARTBEAT
-            self._lasthb = datetime.datetime.now()
-            self._hbwait = int(xmldoc.getElementsByTagName('action')[0].
-                               firstChild.toxml())
-            self.parent.log.debug('ISY HEARTBEAT: ' + self._lasthb.isoformat())
-        if cntrl.format('ST') in msg:  # NODE UPDATE
-            self.parent.nodes._upmsg(xmldoc)
-        elif cntrl.format('_11') in msg:  # WEATHER UPDATE
-            if self.parent.configuration['Weather Information']:
-                self.parent.climate._upmsg(xmldoc)
-        elif cntrl.format('_1') in msg:  # VARIABLE OR PROGRAM UPDATE
-            if '<var' in msg:  # VARIABLE
-                self.parent.variables._upmsg(xmldoc)
-            elif '<id>' in msg:  # PROGRAM
-                self.parent.programs._upmsg(xmldoc)
+        try:
+            cntrl = xmldoc.getElementsByTagName('control')[0].firstChild.toxml()
+            if cntrl == '_0':  # ISY HEARTBEAT
+                self._lasthb = datetime.datetime.now()
+                self._hbwait = int(xmldoc.getElementsByTagName('action')[0].
+                                   firstChild.toxml())
+                self.parent.log.debug('ISY HEARTBEAT: ' + self._lasthb.isoformat())
+            if cntrl == 'ST':  # NODE UPDATE
+                self.parent.nodes._upmsg(xmldoc)
+            if  cntrl[0] != '_': # NODE CONTROL EVENT
+                self.parent.nodes._controlmsg(xmldoc)
+            elif cntrl == '_11':  # WEATHER UPDATE
+                if self.parent.configuration['Weather Information']:
+                    self.parent.climate._upmsg(xmldoc)
+            elif cntrl == '_1':  # VARIABLE OR PROGRAM UPDATE
+                if '<var' in msg:  # VARIABLE
+                    self.parent.variables._upmsg(xmldoc)
+                elif '<id>' in msg:  # PROGRAM
+                    self.parent.programs._upmsg(xmldoc)
+        except:
```

Comment:
This feels awfully dirty. Why the naked except statement here?

Reply:
This... must have been an oversight. I don't remember wrapping that entire block in a naked except, but yeah, it's got to go. I'll fix it.